### PR TITLE
Adding concurrent builds test for benchmark operator

### PR DESCRIPTION
**Pull request by paigerube14**
### Description
This (with addition of benchmark-operator [PR](https://github.com/cloud-bulldozer/benchmark-operator/pull/566)) will add and run the concurrent builds test case 

This is a little different from some of the other scripts in kube-burner because of the need to start **x** number of builds concurrently and get the build and push times. 
Benchmark-operator creates the builds that we can then use to create/start new builds concurrently to get the data for. The benchmark operator makes 1 more namespace then set in the build array to make sure that we can run enough concurrent builds as specified in the build array parameter. 

The results will be outputted into a file named "conc_builds_results.out" and printed out at the end of the run

### CODE REVIEW

**Comment by mffiedler** [comment]
@paigerube14 Please rebase
@chaitanyaenr @mohit-sheth @rsevilla87    Comments?

**Comment by paigerube14** [comment]
I rebased this PR, should be good to review. Thanks!

File: workloads/kube-burner/README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ In order to kick off one of these benchmarks you must use the run.sh script. The
 - **`pod-density`**: `WORKLOAD=pod-density ./run.sh`
 - **`pod-density-heavy`**: `WORKLOAD=pod-density-heavy ./run.sh`
 - **`custom`**: `WORKLOAD=custom ./run.sh`
+- **`concurrent-builds`**
```
Comment (rsevilla87) [inline workloads/kube-burner/README.md:15]:
```suggestion
- **`concurrent-builds`**: `WORKLOAD=concurrent-builds ./run.sh`
```

File: workloads/kube-burner/common.sh [hunk 2]
```diff
@@ -14,11 +14,11 @@ else
   export PROM_TOKEN=$(oc -n openshift-monitoring sa get-token prometheus-k8s)
 fi
 export TOLERATIONS="[{key: role, value: workload, effect: NoSchedule}]"
-export UUID=${UUID:-$(uuidgen)}
+export UUID=${UUID:-$(uuidgen | awk '{print tolower($0)}')}
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:17]: AFAIK uuidgen generates a lowercase uuid 🤔

File: workloads/kube-burner/run.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 . common.sh
 
-remove_update_taint
+#remove_update_taint
```
Comment (rsevilla87) [inline workloads/kube-burner/run.sh:5]: Why did you remove this?, this issue is still present in latest builds

File: workloads/kube-burner/run.sh [hunk 4]
```diff
@@ -86,11 +106,25 @@ if [[ ${PPROF_COLLECTION} == "true" ]] ; then
...
   delete_oldpprof_folder
   get_pprof_secrets
-fi 
-run_workload kube-burner-crd.yaml
+fi
+
+echo "WORKLOAD ${WORKLOAD}"
+if [[ ${WORKLOAD} -eq "concurrent-builds" ]]; then
+   app_array=($APP_LIST)
+   echo "conc builds"
```
Comment (rsevilla87) [inline workloads/kube-burner/run.sh:114]: These two echo statements are unnecessary, (workload name is already printed in L92)

File: workloads/kube-burner/common.sh [hunk 5]
```diff
@@ -177,3 +177,122 @@ remove_update_taint() {
   # This is only here until BZ https://bugzilla.redhat.com/show_bug.cgi?id=2035005 gets resolved
   oc adm taint nodes UpdateInProgress:PreferNoSchedule- --all || true
 }
+
+
+label_node_label() {
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:182]: All this new code is only used by the concurrent builds test, I think we should move it to a new *.sh file and source that file from run.sh, in order to prevent the number of lines of common.sh growing too much.

File: workloads/kube-burner/common.sh [hunk 6]
```diff
@@ -177,3 +177,122 @@ remove_update_taint() {
...
+  colon_param=$(echo $1 | tr "=" ":" | sed 's/:/: /g')
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:183]: Is this colon_param variable actually used?

File: workloads/kube-burner/common.sh (hunk 2 above)
Comment (paigerube14) [reply workloads/kube-burner/common.sh:17]: I'm not sure if it was lowercase on my mac, but a lot of the other commands I can't run. So I'll remove

File: workloads/kube-burner/common.sh (hunk 6 above)
Comment (paigerube14) [reply workloads/kube-burner/common.sh:183]: No it currently is not but I was trying to work towards combining this function and label_nodes to be more generic to take in the wanted label and export the correct parameters with a colon or with equals. Will update if I finalize it

File: workloads/kube-burner/run.sh (hunk 3 above)
Comment (paigerube14) [reply workloads/kube-burner/run.sh]: took out during my own testing, adding back

File: workloads/kube-burner/build_helper.sh [hunk 7]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/bash
+
+. common.sh
```
Comment (rsevilla87) [inline workloads/kube-burner/build_helper.sh:3]: we snouldn't source common again, as it's already sourced by run.sh

File: workloads/kube-burner/common.sh [hunk 8]
```diff
@@ -195,3 +173,39 @@ remove_update_taint() {
   # This is only here until BZ https://bugzilla.redhat.com/show_bug.cgi?id=2035005 gets resolved
   oc adm taint nodes UpdateInProgress:PreferNoSchedule- --all || true
 }
+
+
+label_node_with_label() {
+  colon_param=$(echo $1 | tr "=" ":" | sed 's/:/: /g')
+  export POD_NODE_SELECTOR="{$colon_param}"
+  export NODE_SELECTOR="$1"
+  echo $NODE_SELECTOR
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:182]: Can you replace all the echo statements by log?

Comment (paigerube14) [reply workloads/kube-burner/common.sh]: done

File: workloads/kube-burner/common.sh [hunk 9]
```diff
@@ -195,3 +173,39 @@ remove_update_taint() {
   # This is only here until BZ https://bugzilla.redhat.com/show_bug.cgi?id=2035005 gets resolved
   oc adm taint nodes UpdateInProgress:PreferNoSchedule- --all || true
 }
+
+
+label_node_with_label() {
+  colon_param=$(log $1 | tr "=" ":" | sed 's/:/: /g')
+  export POD_NODE_SELECTOR="{$colon_param}"
+  export NODE_SELECTOR="$1"
+  log $NODE_SELECTOR
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:182]: We don't need this one

File: workloads/kube-burner/common.sh [hunk 10]
```diff
@@ -195,3 +173,39 @@ remove_update_taint() {
...
+  NODE_COUNT=$(oc get node -o name --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker= | wc -l | xargs )
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:183]: we don't need xargs here

File: workloads/kube-burner/common.sh [hunk 11]
```diff
@@ -195,3 +174,37 @@ remove_update_taint() {
...
+  colon_param=$(log $1 | tr "=" ":" | sed 's/:/: /g')
+  export POD_NODE_SELECTOR="{$colon_param}"
+  export NODE_SELECTOR="$1"
+  NODE_COUNT=$(oc get node -o name --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker= | wc -l )
+  if [[ ${NODE_COUNT} -le 0 ]]; then
+    log "Node count <= 0: ${NODE_COUNT}"
+    exit 1
+  fi
+  export WORKER_NODE_NAMES=$(oc get node -o name --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker= | head -n ${NODE_COUNT})
+  if [[ $(log "${WORKER_NODE_NAMES}" | wc -l) -lt ${NODE_COUNT} ]]; then
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:189]: I think you mistakenly replaced echo by log here

File: workloads/kube-burner/common.sh [hunk 12]
```diff
@@ -195,3 +174,37 @@ remove_update_taint() {
...
+label_node_with_label() {
+  colon_param=$(log $1 | tr "=" ":" | sed 's/:/: /g')
+  export POD_NODE_SELECTOR="{$colon_param}"
+  export NODE_SELECTOR="$1"
+  NODE_COUNT=$(oc get node -o name --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker= | wc -l )
+  if [[ ${NODE_COUNT} -le 0 ]]; then
+    log "Node count <= 0: ${NODE_COUNT}"
+    exit 1
+  fi
+  export WORKER_NODE_NAMES=$(oc get node -o name --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker= | head -n ${NODE_COUNT})
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:188]: we don't need to export this variable

File: workloads/kube-burner/common.sh [hunk 13]
```diff
@@ -195,3 +174,37 @@ remove_update_taint() {
   # This is only here until BZ https://bugzilla.redhat.com/show_bug.cgi?id=2035005 gets resolved
   oc adm taint nodes UpdateInProgress:PreferNoSchedule- --all || true
 }
+
+
+label_node_with_label() {
+  colon_param=$(log $1 | tr "=" ":" | sed 's/:/: /g')
```
Comment (rsevilla87) [inline workloads/kube-burner/common.sh:180]: I think you mistakenly replaced echo by log here

File: workloads/kube-burner/common.sh (hunk 12 above)
Comment (paigerube14) [reply workloads/kube-burner/common.sh:188]: @rsevilla87 if i don't need to export, should I put to variable to lower case?

**Comment by paigerube14** [comment]
@rsevilla87 any other updates needed or can we get this merged?
